docs/azure/sdk/snippets/unit-testing/ResourceManager/ArmClient.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System.ComponentModel;
+using System.Collections.Concurrent;
+
+#nullable disable
+
+namespace UnitTestingSampleApp.ResourceManager;
+
+public partial class ArmClient
+{
+    private readonly ConcurrentDictionary<Type, object> _clientCache = new();
+
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public virtual T GetCachedClient<T>(Func<ArmClient, T> clientFactory)
+            where T : class
+    {
+        return _clientCache.GetOrAdd(typeof(T), type => clientFactory(this)) as T;
```

Comment:
Use a direct cast in this case, especially when nullable is enabled. Otherwise, it's treated as potentially null.

```suggestion
        return (T)_clientCache.GetOrAdd(typeof(T), type => clientFactory(this));
```